File: AS-2025-2026/4X-TPSIT/es-00_2025-11-14_ProcessGreetingSimpsons/es00.c
/*
This code generates a parent process and a child process. Depending on the execution mode, the PID is printed.
*/

#include <stdio.h>
#include <unistd.h>

int main() {
	int pid = fork();
	if(pid == 0) {
		printf("child process [%d] >> main process [%d]\n", getpid(), getppid());
	} else {
		printf("main process [%d] >> child process [%d]\n", getpid(), pid);
	}
	return 0;
}

File: AS-2025-2026/4X-TPSIT/es-00_2025-11-14_ProcessGreetingSimpsons/es00.py
# This code generates a parent process and a child process.
# Depending on the execution mode, the PID is printed.

#!/usr/bin/python3
from os import fork, getpid, getppid

if __name__ == '__main__':
	pid = fork()
	if pid == 0:
		print(f"child process [{getpid()}] >> main process [{getppid()}]")
	else:
		print(f"main process [{getpid()}] >> child process [{pid}]")